frontend/packages/console-app/src/components/quick-starts/QuickStartDrawer.tsx

```diff
@@ -99,11 +95,43 @@ const QuickStartDrawer: React.FC<QuickStartDrawerProps> = ({
   ) : null;
 
   return (
-    <Drawer isExpanded={!!activeQuickStartID} isInline>
-      <DrawerContent panelContent={panelContent}>
-        <DrawerContentBody className="co-quick-start-drawer__body">{children}</DrawerContentBody>
-      </DrawerContent>
-    </Drawer>
+    <>
+      <Drawer isExpanded={!!activeQuickStartID} isInline>
+        <DrawerContent panelContent={panelContent}>
+          <DrawerContentBody className="co-quick-start-drawer__body">{children}</DrawerContentBody>
+        </DrawerContent>
+      </Drawer>
+      <Modal
+        className="co-quick-start-drawer__modal"
+        isOpen={modalOpen}
+        variant={ModalVariant.small}
+        showClose={false}
+        title="Are you sure you want to leave the tour?"
+        footer={
+          <Flex>
+            <FlexItem align={{ default: 'alignRight' }}>
+              <Button variant="secondary" onClick={() => setModalOpen(false)}>
+                Cancel
+              </Button>
+            </FlexItem>
+            <FlexItem>
+              <Button
+                variant="primary"
+                onClick={() => {
+                  setModalOpen(false);
+                  onClose();
+                }}
+              >
+                Leave
+              </Button>
+            </FlexItem>
+          </Flex>
+        }
+        isFullScreen
+      >
+        {"Any progress you've made will be saved."}
+      </Modal>
```

Comment:
Maybe extract this out into a new component?

Reply:
updated